content/en/docs/tasks/traffic-management/egress/egress-kubernetes-services/index.md

```diff
@@ -0,0 +1,268 @@
+    for `httpbin.org`:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    kind: Service
+    apiVersion: v1
+    metadata:
+      name: my-httpbin
+    spec:
+      type: ExternalName
+      externalName: httpbin.org
+      ports:
+      - name: http
+        protocol: TCP
+        port: 80
+    EOF
+    {{< /text >}}
+
+1.  Observe your service. Note that it does not have a cluster IP.
+
+    {{< text bash >}}
+    $ kubectl get svc my-httpbin
+    NAME         TYPE           CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+    my-httpbin   ExternalName   <none>       httpbin.org   80/TCP    4s
+    {{< /text >}}
+
+1.  Access `httpbin.org` via the Kubernetes service's hostname from the source pod without Istio sidecar:
+
+    {{< text bash >}}
+    $ kubectl exec -it $SOURCE_POD_WITHOUT_ISTIO -n without-istio -c sleep -- curl my-httpbin.default.svc.cluster.local/headers
+    {
+      "headers": {
+        "Accept": "*/*",
+        "Host": "my-httpbin.default.svc.cluster.local",
+        "User-Agent": "curl/7.55.0"
+      }
+    }
+    {{< /text >}}
+
+1.  In this example, unencrypted HTTP requests are sent to `httpbin.org`. For the sake of the example only, you disable
+    the TLS mode and allow the unencrypted traffic to the external service. In the real life scenarios, we recommend
+    to perform [Egress TLS origination](/docs/tasks/traffic-management/egress/egress-tls-origination/) by Istio.
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: DestinationRule
+    metadata:
+      name: my-httpbin
+    spec:
+      host: my-httpbin.default.svc.cluster.local
+      trafficPolicy:
+        tls:
+          mode: DISABLE
+    EOF
+    {{< /text >}}
+
+1.  Access `httpbin.org` via the Kubernetes service's hostname from the source pod with Istio sidecar. Notice the
+    headers added by Istio sidecar, for example, `X-Istio-Attributes` and `X-Envoy-Decorator-Operation`. Also note that
+    the `Host` header equals to your service's hostname.
+
+    {{< text bash >}}
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl my-httpbin.default.svc.cluster.local/headers
+    {
+      "headers": {
+        "Accept": "*/*",
+        "Host": "my-httpbin.default.svc.cluster.local",
+        "User-Agent": "curl/7.55.0",
+        "X-B3-Sampled": "0",
+        "X-B3-Spanid": "5b68b3f953945a08",
+        "X-B3-Traceid": "0847ba2513aa0ffc5b68b3f953945a08",
+        "X-Envoy-Decorator-Operation": "my-httpbin.default.svc.cluster.local:80/*",
+        "X-Istio-Attributes": "CigKGGRlc3RpbmF0aW9uLnNlcnZpY2UubmFtZRIMEgpteS1odHRwYmluCioKHWRlc3RpbmF0aW9uLnNlcnZpY2UubmFtZXNwYWNlEgkSB2RlZmF1bHQKOwoKc291cmNlLnVpZBItEitrdWJlcm5ldGVzOi8vc2xlZXAtNjZjOGQ3OWZmNS04aG1neC5kZWZhdWx0CkAKF2Rlc3RpbmF0aW9uLnNlcnZpY2UudWlkEiUSI2lzdGlvOi8vZGVmYXVsdC9zZXJ2aWNlcy9teS1odHRwYmluCkIKGGRlc3RpbmF0aW9uLnNlcnZpY2UuaG9zdBImEiRteS1odHRwYmluLmRlZmF1bHQuc3ZjLmNsdXN0ZXIubG9jYWw="
+      }
```

Comment:
would be good to explain these 2 headers a bit... what do they do and how they help?

Reply:
Explaining these headers will provide unneeded information to the reader. Both headers are implementation details:

1. https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter#x-envoy-decorator-operation

> If this header is present on ingress requests, its value will override any locally defined operation (span) name on the server span generated by the tracing mechanism. Similarly, if this header is present on an egress response, its value will override any locally defined operation (span) name on the client span.

1. https://github.com/istio/proxy/commit/3781be0e624470a4ee613c4ffc9816976f5dd701

x-istio-attributes is used for setting hash policy for the [Maglev load balancing of Envoy](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#maglev)

I just mention the headers to demonstrate the reader that the request indeed passed through the sidecar proxy.